src/cplus_plugin/gui/implementation_model_editor_dialog.py

```diff
@@ -186,6 +205,7 @@ def _create_implementation_model(self):
             self._implementation_model.description = self.txt_description.toPlainText()
 
         self._layer = self._get_selected_map_layer()
+        self._implementation_model.fill_style = self.fill_symbol_layer().properties()
```

Comment:
Even though this is unlikely but lets add a check if the `fill_symbol_layer` functions returns a non-null layer before accessing its `properties`.

Reply:
By default, the button does not allow nulls

![image](https://github.com/kartoza/cplus-plugin/assets/3840267/0d9505a5-e76d-4fcc-b1c2-b3f25c913923)

But am also checking if there is at least one fill symbol layer in the symbol before saving in the implementation model dialog.

Reply:
@gkahiu I meant to have similar check as [this](https://github.com/kartoza/cplus-plugin/pull/268/files#diff-38d09860f74cb1d4956e2ab76f02c0c9d42348ce2380a82ac227ab97c4b39062R184) before accessing the `properties` function. If you look at line 247 you could see that there a potential for the function `fill_symbol_layer` to return `None`.

Reply:
Hi @Samweli , see the updated PR. I have included the check.